src/model/Group.js

```diff
@@ -27,13 +29,23 @@ class Group extends Step {
    */
   constructor(name, type, config = {}) {
     /* Clean up config agains input or output ports or bindings*/
-    delete config.i;
-    delete config.o;
+    if (!(config instanceof Action)) {
+      if (_.has(config, 'i') || _.has(config, 'o')) {
+        throw new Error('Group could not have any ports');
+      }
+    } else {
+      throw new Error('Group could be created only using config object');
+    }
 
     super(name, config);
 
     this.type = type || 'default';
   }
+
+  /* Override the parent function to prohibit the port addition*/
+  // eslint-disable-next-line class-methods-use-this
+  _onActionChanged() {
+  }
```

Comment:
Does this code really prevents port addition? I think, an Action will still hold new ports but they will be out of sync with the Step. That's not good.